babel/core.py

```diff
@@ -263,7 +263,7 @@ def negotiate(
     @classmethod
     def parse(
         cls,
-        identifier: str | Locale | None,
+        identifier: str | Locale,
```

Comment:
It's a bit complicated here since all the LC_... constants could be None if there is a problem determining the default and they are potentially passed to this method, meaning for type checkers, it sees this function as accepting None, but always raising the exception here.

Reply:
Great points, thanks! Updated the approach here to docs-only.